### ISSUE TITLE
Map formatting badly in some cases

### ISSUE THREAD

**Issue by Villar74** [kkinnear/zprint#133]
Here is my config:
```
:zprint {:style :community
                           :width 100
                           :parallel? true
                           :fn-map {"reg-event-fx" :arg1-force-nl
                                    "reg-event-db" :arg1-force-nl
                                    "reg-sub" :arg1-force-nl
                                    "reg-fx" :arg1-force-nl
                                    "if" :arg1-force-nl}
                           :map {:comma? false
                                 :hang? true
                                 :indent 0
                                 :sort? true
                                 :force-nl? true}
                           :vector {:respect-nl? true
                                    :respect-bl? true
                                    :indent-only? true}
                           :cache {:directory ".zprint"
                                   :location "HOME"}
                           :old? false}}
```
I formatting project and get this unexpected indents:
![example](https://imgur.com/JKMclcm.png)

**Comment by kkinnear** [comment]
Thanks for noticing this.  You have run into something that I noticed in a different context.  The root problem here is that `:indent-only?` doesn't work well with other approaches.  I have created an issue explicitly for that problem, #134.  I've explained this issue a bit there, but the basic idea is that if you try to format a vector as `:indent-only?` and it is inside of something that isn't formatted as `:indent-only?`, it doesn't work.  Indent-only can only be used throughout the formatting at present.   I'm on the fence about trying to fix this and just making it "how it works", since `:respect-nl?` is so similar to `:indent-only?`.   But I haven't yet decided.  Right now I'm working on the other issue you submitted, #132.  I'm reworking the entire way comments are handled everywhere, which if it works, will simplify things a lot.  But it is a huge change.

I would suggest that using `:respect-nl?` inside the vector would most likely give you what you are looking for -- it is similar to `:indent-only?`, but it will add newlines in order to stay within the width.  More importantly, it will play well with the formatting used for maps and everything else.  Also, `:respect-bl?` is a subset of `:respect-nl?`.  Pick one or the other -- using both means that you are using `:respect-nl?`.  I would suggest this for a your options map:
```
 :zprint {:style :community
                           :width 100
                           :parallel? true
                           :fn-map {"reg-event-fx" :arg1-force-nl
                                    "reg-event-db" :arg1-force-nl
                                    "reg-sub" :arg1-force-nl
                                    "reg-fx" :arg1-force-nl
                                    "if" :arg1-force-nl}
                           :map {:comma? false
                                 :hang? true
                                 :indent 0
                                 :sort? true
                                 :force-nl? true}
                           :vector {:respect-nl? true}
                           :cache {:directory ".zprint"
                                   :location "HOME"}
                           :old? false}}
```
That will give you almost the same output as `:vector {:indent-only? true}`, except that it will wrap lines in vectors that are over 100 characters.

Another thought, you might investigate `{:style :hiccup}`, which at least one person thinks makes hiccup vectors look better.  I didn't see any positive difference in your example, but that doesn't mean that you wouldn't find it useful for larger things.

One more comment -- thanks again for your configuration, it helps a lot and allows me to reproduce things without a lot of questions back and forth.  The other thing that would help me is if you cut and pasted the examples of what didn't work instead of using a link.  I was not able to cut the example structure out of the link, and had to type it in from scratch.  Which, besides being a pain, is also slightly error prone.  Thanks!
